### PR TITLE
Adding Teamview Method only in Pigskin.

### PR DESCRIPTION
**Pull request by kaileu**
I decided to make another PR  but only for Pigskin.
How the GUI should Looks like is independant from the working function in Pigskin.

### CODE REVIEW

File: resources/lib/pigskin.py [hunk 1]
```diff
@@ -228,6 +228,31 @@ def get_weeks_games(self, season, season_type, week):
...
+                for conference in teams['modules']:
+                    if 'content' in teams['modules'][conference]:
+                        for teamname in teams['modules'][conference]['content']:
+                            if team == teamname['fullName']:
+                                team = teamname['seoname']
+
+                url = self.config['modules']['ROUTES_DATA_PROVIDERS']['team_detail'].replace(':team', team)
+                games_data = self.make_request(url, 'get')
+                # collect the games from all keys in 'modules' for a specific season
+                games = [g for x in games_data['modules'].keys() if x == 'videos'+season for g in games_data['modules'][x]['content']]
```
Comment (aqw) [inline resources/lib/pigskin.py]: Does this line pass PEP8? Specifically `'videos'+season`

File: resources/lib/pigskin.py [hunk 2]
```diff
@@ -228,6 +228,31 @@ def get_weeks_games(self, season, season_type, week):
...
+            if team is None:
+                return teams
+            else:
+                #Looking for the Teamname
+                for conference in teams['modules']:
+                    if 'content' in teams['modules'][conference]:
+                        for teamname in teams['modules'][conference]['content']:
+                            if team == teamname['fullName']:
+                                team = teamname['seoname']
+
```
Comment (aqw) [inline resources/lib/pigskin.py]: What if the requested team is never found?

File: resources/lib/pigskin.py [hunk 3]
```diff
@@ -228,6 +228,31 @@ def get_weeks_games(self, season, season_type, week):
...
         return sorted(games, key=lambda x: x['gameDateTimeUtc'])
+        
+    def get_team_games(self, season, team=None):
+        try:
+            url = self.config['modules']['ROUTES_DATA_PROVIDERS']['teams']
+            teams = self.make_request(url, 'get')
+            if team is None:
+                return teams
+            else:
+                #Looking for the Teamname
```
Comment (aqw) [inline resources/lib/pigskin.py]: Nitpick: PEP8

Comment (kaileu) [inline resources/lib/pigskin.py]: i don't know what you want to tell me :-D the pep8 checker only says that the lines are too Long but i think thats fine.

File: resources/lib/pigskin.py (hunk 1 above)
Comment (kaileu) [inline resources/lib/pigskin.py]: as mentioned above the online checker says ist ok

File: resources/lib/pigskin.py (hunk 2 above)
Comment (kaileu) [inline resources/lib/pigskin.py]: your right that should be an exception.
I thought it would be ctached in the gui part but your right.

**Comment by kaileu** [comment]
to your last comment i think it would be a bad idea to Change only some functions.
I would clean it up all together so thats the Output is equivalent to other functions.

File: resources/lib/pigskin.py (hunk 2 above)
Comment (aqw) [inline resources/lib/pigskin.py]: It should explicitly return None or something like that.

File: resources/lib/pigskin.py (hunk 3 above)
Comment (aqw) [inline resources/lib/pigskin.py]: Sorry, I should have been more specific: comments usually have a space following the "#"

As I said, it's a nitpick and something I can fix when merging.

**Comment by aqw** [comment]
@kaileu Add the explicit None return, and then I'll happily merge this. :-)

I'm fine with not setting this function up to return the data in a standard format. but we're definitely not going to convert the entire code-base in one huge commit. It'll take time and we'll move through one function at a time, updating both default.py and pigskin.py in tandem. That will help significantly in controlling regressions.

Thanks again for this PR, and I really like tackling the pigskin.py interface first before doing the GUI. It helps keep the problem-space focused.

---Alex

**Comment by kaileu** [comment]
It should be Cleaner now :-)

File: resources/lib/pigskin.py [hunk 4]
```diff
@@ -228,6 +228,33 @@ def get_weeks_games(self, season, season_type, week):
...
+            teams = self.make_request(url, 'get')
+            if team is None:
+                return teams
+            else:
+                # Looking for the Teamname
+                for conference in teams['modules']:
+                    if 'content' in teams['modules'][conference]:
+                        for teamname in teams['modules'][conference]['content']:
+                            if team == teamname['fullName']:
+                                team = teamname['seoname']
```
Comment (emilsvennesson) [inline resources/lib/pigskin.py]: You should break out of the loop when the team is found.

Comment (kaileu) [inline resources/lib/pigskin.py]: its a break at this Point as the innerloop is 
> if 'content' in teams['modules'][conference]: 

am i right ?

Comment (emilsvennesson) [inline resources/lib/pigskin.py]: Yeah, as soon as the team name is found the loop has served its purpose. Basically, you should add the ```break``` statement after line 244. :-)

Comment (kaileu) [inline resources/lib/pigskin.py]: added :-)

**Comment by aqw** [comment]
@kaileu Looks good to me. Thanks for adding this. :-)
@emilsvennesson Thanks for catching the missing `break`. I missed that.

**Pull request closed without merging**